pkg/ui/static/js/graph.js

```diff
@@ -521,7 +530,11 @@ Prometheus.Graph.prototype.submitQuery = function() {
   var startTime = new Date().getTime();
   var rangeSeconds = self.parseDuration(self.rangeInput.val());
   var resolution = parseInt(self.queryForm.find("input[name=step_input]").val()) || Math.max(Math.floor(rangeSeconds / 250), 1);
-  var maxSourceResolution = self.queryForm.find("select[name=max_source_resolution_input]").val();
+  var maxSourceResolution = self.maxSourceResolutionInput.val()
+  // normalize max_source_resolution parameter
+  if ( maxSourceResolution == "auto") {
```

Comment:
Why not instead supporting `auto` value in Thanos Querier server side? :thinking:

Reply:
Yes I like the idea.
Should I add this to this PR or open another PR ?

Reply:
Also remember that grafana users can't set this parameter and probably want the "auto" parameter as default.